### ISSUE TITLE
Nerves.Network.teardown/1 doesn't tear down

### ISSUE THREAD

**Issue by fhunleth** [nerves-project/nerves_network#22]
I was expecting `teardown/1` to disassociate from the access point and cleanup the IP addresses assigned to the link. I.e., restore the link to what it looked like before `setup/2` was called. That's not happening. Here's what I did:

```
iex> Nerves.Network.setup "wlan0", stuff...

# wait for association, DHCP, etc.

iex> Nerves.Network.teardown "wlan0"
[debug] Elixir.Nerves.Network teardown(wlan0)
{:ok, {%{}, %{}}}

# wait a few seconds (shouldn't be necessary, but I did just in case)

iex> cmd("ip addr")
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: usb0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether ea:e4:ff:c9:0a:5e brd ff:ff:ff:ff:ff:ff
    inet 169.254.112.52/16 brd 169.254.255.255 scope global usb0
       valid_lft forever preferred_lft forever
    inet6 fe80::e8e4:ffff:fec9:a5e/64 scope link 
       valid_lft forever preferred_lft forever
3: sit0@NONE: <NOARP> mtu 1480 qdisc noop qlen 1
    link/sit 0.0.0.0 brd 0.0.0.0
4: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether b8:27:eb:d1:55:f0 brd ff:ff:ff:ff:ff:ff
    inet 192.168.11.116/24 brd 192.168.11.255 scope global wlan0
       valid_lft forever preferred_lft forever
    inet6 fe80::ba27:ebff:fed1:55f0/64 scope link 
       valid_lft forever preferred_lft forever
```

As you can see `wlan0` still has an IP address.

**Comment by adkron** [comment]
I'm having the same issue. I started looking into the matter. When we try to remove the interface from `SystemRegistry`, the interface remains in the registry. I pulled down `SystemRegistry` to make sure that it works with strings and that seems to work fine according to my testing. I now believe that our interaction with the registry might not be right, I'm digging in there now.

**Comment by metelik** [comment]
I think another issue I had logged might be related:  [https://github.com/nerves-project/nerves_network/issues/49](url) I am trying to fix it with adkron.

**Comment by adkron** [comment]
You cannot teardown  `:default` interface through `Nerves.Network.teardown/1`. `Nerves.Network.teardown/1` uses priority `:nerves_nerwork` and defaults are started under priority `:default`.

I'm not sure if that is desired or related, but that is what I was initially talking about in this issue. My original hypothesis was wrong, but that was due to the confusing nature of `default`.

If this is the desired behaviour, we should document it. If this is not the desired behaviour then what is? Thoughts?

**Comment by ConnorRigby** [comment]
Fixed in #97
